src/integrator/sycl/mklsycl_extensions.cpp

```diff
@@ -61,16 +76,19 @@ namespace blas  {
     void hadamard_product(cl::sycl::queue *syclQue, int M, int N, const T *A, int LDA,
                           T *B, int LDB) {
 
-        cl::sycl::range<2> threads(16, 16);
-        cl::sycl::range<2> blocks(util::div_ceil(M, threads[0]),
-                                  util::div_ceil(N, threads[1]));
+        cl::sycl::range<3> threads(16, 16, 1);
```

Comment:
This should be 2D, no?

Reply:
Addressed. I have lost this one too from reverts.